src/test/java/org/graylog/plugins/pagerduty/client/PagerDutyTest.java

```diff
@@ -0,0 +1,144 @@
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.graylog.events.notifications.EventNotificationContext;
+import org.graylog.plugins.pagerduty.PagerDutyNotificationConfig;
+import org.graylog.plugins.pagerduty.dto.PagerDutyMessage;
+import org.graylog.plugins.pagerduty.dto.PagerDutyResponse;
+import org.graylog2.streams.StreamService;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.slf4j.Logger;
+
+/**
+ * @author Edgar Molina
+ *
+ */
+public class PagerDutyTest
+{
+    @Mock
+    private StreamService streamServiceMock;
+    @Mock
+    private PagerDutyNotificationConfig configMock;
+    @Mock
+    private CloseableHttpClient httpClientMock;
+    @Mock
+    private ObjectMapper objectMapperMock;
+    @Mock
+    private MessageFactory messageFactoryMock;
+    @Mock
+    private Logger loggerMock;
+    @Mock
+    private PagerDutyMessage messageMock;
+    @Mock
+    private EventNotificationContext contextMock;
+    @Mock
+    private CloseableHttpResponse httpResponseMock;
+    @Mock
+    private HttpEntity entityMock;
+    @Mock
+    private InputStream contentMock;
+    @Mock
+    private PagerDutyResponse pagerDutyResponseMock;
+
+
+    private PagerDuty sut;
+
+    @Before
+    public void setUp() throws ClientProtocolException, IOException
+    {
+        MockitoAnnotations.initMocks(this);
+        when(messageFactoryMock.createTriggerMessage(contextMock)).thenReturn(messageMock);
+        when(objectMapperMock.writeValueAsString(messageMock)).thenReturn("{test='json'}");
+        when(httpClientMock.execute(any(HttpPost.class))).thenReturn(httpResponseMock);
+        when(entityMock.getContent()).thenReturn(contentMock);
+        when(httpResponseMock.getEntity()).thenReturn(entityMock);
+        when(objectMapperMock.readValue(contentMock, PagerDutyResponse.class))
+            .thenReturn(pagerDutyResponseMock);
+        sut =
+            new PagerDuty(
+                streamServiceMock,
+                configMock,
+                httpClientMock,
+                objectMapperMock,
+                messageFactoryMock,
+                loggerMock);
+    }
+
+    @Test
+    public void testSuccessfulTriggerAttempt() throws IOException
+    {
+        // Execute
+        PagerDutyResponse result = sut.trigger(contextMock);
+
+        // Assert
+        assertEquals("Wrong Response Object", pagerDutyResponseMock, result);
+        verify(httpResponseMock).close();
+        verify(loggerMock).debug("Triggering event in PagerDuty with context: {}", contextMock);
+        verify(loggerMock).debug("Request Payload: {}", "{test='json'}");

```

Comment:
Nit: Just my own opinion, but verifying debug logs seems like overkill and the kind of thing that might make your tests fragile.